test/services/licences/fetch-licence-returns.service.test.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const DatabaseSupport = require('../../support/database.js')
+const ReturnLogHelper = require('../../support/helpers/return-log.helper.js')
+
+// Thing under test
+const FetchLicenceReturnsService = require('../../../app/services/licences/fetch-licence-returns.service')
+const LicenceHelper = require('../../support/helpers/licence.helper')
+
+describe('Fetch licence returns service', () => {
+  const licenceId = 'fef693fd-eb6f-478d-9f79-ab24749c5dc6'
+
+  beforeEach(async () => {
+    await DatabaseSupport.clean()
+  })
+
+  describe('when there is no optional data in the model', () => {
+    let firstReturn
+    let latestReturn
```

Comment:
I could be wrong but it looks like we are saving to these so we can access the IDs for asserting in the tests.

We have 2 alternate strategies for this. The first and easiest is just to specify the ID when you create the record

```javascript
await ReturnLogHelper.add({ ...returnData, id: '72c01307-1a63-48db-b59d-021e7bf623c6' })
```

The second is to ignore them on the basis we clear the table before inserting, so we know there are only these records in the table.

```javascript
  expect(result).to.equal({}, { skip: ['id] }
```

So, I think you can for go them. Else, we like to specify any `let`s _after_ the `const`s with a good ol' newline in between!

Reply:
Code updated.

This is needed:

```
      firstReturn.licenceRef = license.licenceRef
      latestReturn.licenceRef = license.licenceRef
```

As the licenceRef can't be overidden and forces `generateLicenceRef()`